spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/FastForwardBranchProcedure.java

```diff
@@ -77,9 +77,9 @@ public InternalRow[] call(InternalRow args) {
     return modifyIcebergTable(
         tableIdent,
         table -> {
-          long currentRef = table.currentSnapshot().snapshotId();
+          long currentRef = table.snapshot(source).snapshotId();
```

Comment:
wouldn't we fail here if that particular branch doesn't exist? Because the branch existence checks only happen one step later in `table.manageSnapshots().fastForwardBranch`. I think we should also add a test where that procedure is called with a non-existing branch

Reply:
+1, looks like we would hit another NPE in this case.

Reply:
Definitley here's a NPE bug here, good catch. I'll add a test.

Reply:
I updated. I added a test for 2 cases, 1 where the branch to fast forward does not exist and 1 for where the "to" branch does not exist (the second case is handled by the API itself but I think in this case it's worth making sure the output is what one would expect).